NumLib/TimeStepping/Algorithms/IterationNumberBasedTimeStepping.cpp

```diff
@@ -19,31 +20,33 @@
 
 namespace NumLib
 {
-IterationNumberBasedAdaptiveTimeStepping::
-    IterationNumberBasedAdaptiveTimeStepping(
-        double t0, double tn, double min_ts, double max_ts, double initial_ts,
-        std::vector<std::size_t>& iter_times_vector,
-        std::vector<double>& multiplier_vector)
-    : TimeStepAlgorithm(t0, tn),
-      _iter_times_vector(iter_times_vector),
-      _multiplier_vector(multiplier_vector),
-      _min_ts(min_ts),
-      _max_ts(max_ts),
-      _initial_ts(initial_ts),
-      _max_iter(_iter_times_vector.empty() ? 0 : _iter_times_vector.back()),
-      _iter_times(0),
-      _n_rejected_steps(0)
+IterationNumberBasedTimeStepping::IterationNumberBasedTimeStepping(
+    double const t_initial, double const t_end, double const min_dt,
+    double const max_dt, double const initial_dt,
+    std::vector<int>&& iter_times_vector,
+    std::vector<double>&& multiplier_vector)
+    : TimeStepAlgorithm(t_initial, t_end),
+      _iter_times_vector(std::move(iter_times_vector)),
+      _multiplier_vector(std::move(multiplier_vector)),
+      _min_dt(min_dt),
+      _max_dt(max_dt),
+      _initial_dt(initial_dt),
+      _max_iter(_iter_times_vector.empty() ? 0 : _iter_times_vector.back())
```

Comment:
Sorry, I'm not so familiar with the time stepping. Could you give a simple and short example what data is stored in  `_iter_times_vector`?